### PR TITLE
[sample] Pitch calculating refactoring

### PR DESCRIPTION
**Pull request by IlyaTikhanov**
When it is 16K resolution and 444 Chroma subsampling
we need value more then 16 bit variable allows
So need to use PitchLow and PitchHigh instead of just Pitch

Issue: MDP-58640

### CODE REVIEW

File: samples/sample_common/src/sysmem_allocator.cpp [hunk 1]
```diff
@@ -107,87 +107,102 @@ mfxStatus SysMemFrameAllocator::LockFrame(mfxMemId mid, mfxFrameData *ptr)
...
-        ptr->Pitch = 2 * Width2;
+        ptr->PitchHigh = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) / (1 << 16));
+        ptr->PitchLow = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) % (1 << 16));
         break;
 #if (MFX_VERSION >= 1028)
     case MFX_FOURCC_RGB565:
         ptr->G = ptr->B;
         ptr->R = ptr->B;
-        ptr->Pitch = 2 * Width2;
+        ptr->PitchHigh = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) >> 16);
```
Comment (mgonchar) [inline samples/sample_common/src/sysmem_allocator.cpp]: please, use one of the approaches, either
```
ptr->PitchHigh = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) / (1 << 16));
ptr->PitchLow = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) % (1 << 16));
```
or
```
ptr->PitchHigh = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) >> 16);
ptr->PitchLow = (mfxU16)((2 * MSDK_ALIGN32(fs->info.Width)) & 0xffff);
```

File: samples/sample_common/src/sysmem_allocator.cpp [hunk 2]
```diff
@@ -107,87 +107,102 @@ mfxStatus SysMemFrameAllocator::LockFrame(mfxMemId mid, mfxFrameData *ptr)
...
-        ptr->Pitch = 3 * Width2;
+        ptr->PitchHigh = (mfxU16)((3 * MSDK_ALIGN32(fs->info.Width)) / (1 << 16));
+        ptr->PitchLow = (mfxU16)((3 * MSDK_ALIGN32(fs->info.Width)) % (1 << 16));
         break;
 #if !(defined(_WIN32) || defined(_WIN64))
     case MFX_FOURCC_RGBP:
         ptr->G = ptr->B + Width2 * Height2;
         ptr->R = ptr->B + Width2 * Height2 * 2;
-        ptr->Pitch = Width2;
+        ptr->PitchHigh = (mfxU16)((3 * MSDK_ALIGN32(fs->info.Width)) / (1 << 16));
```
Comment (mgonchar) [inline samples/sample_common/src/sysmem_allocator.cpp]: why ```3 * MSDK_ALIGN32```?

File: samples/sample_common/src/sysmem_allocator.cpp (hunk 1 above)
Comment (IlyaTikhanov) [reply samples/sample_common/src/sysmem_allocator.cpp]: Now only the 1st approach is used

File: samples/sample_common/src/sysmem_allocator.cpp (hunk 2 above)
Comment (IlyaTikhanov) [reply samples/sample_common/src/sysmem_allocator.cpp]: Changed. We don't need to multiply by 3.

**Pull request merged**